### ISSUE TITLE
What's the parameters passed to closures?

### ISSUE THREAD

**Issue by shaitao** [tokio-rs/doc-push#64]
I saw in examples like
.map(|_|{...})
.and_then(|(A,B)|{...})
.and_then(|(A,_)|{...})
.and_then(|x|{...})
...
It's hard to find what is passed to the closure.

**Comment by rawrmonstar** [comment]
I believe the closures accept the result value of the previous future.

**Comment by shaitao** [comment]
> 
> 
> I believe the closures accept the result value of the previous future.

yes it is, but many funcitons return future and Self::Item changes around, it's hard to find.

**Comment by carllerche** [comment]
Related #64

**Comment by carllerche** [comment]
We should be sure to answer this question as part of the combinator section in https://github.com/tokio-rs/doc-push/blob/master/outline/futures-streams-sinks.md#contents-1.
